mautic/mautic
# Refactored building config bundle metadata

**Please be sure you are submitting this against the _staging_ branch.**

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | Y
| New feature? | 
| Automated tests included? | Y
| Related user documentation PR URL | 
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | 
| BC breaks? | 
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

[//]: # ( Required: )
#### Description:
This refactors the code in app/config/config.php that built bundle metadata. It organizes into classes so that tests can be written. 

It also fixes a regression issue introduced by https://github.com/mautic/mautic/pull/8265 that was originally fixed https://github.com/mautic/mautic/pull/8366.

[//]: # ( As applicable: )
#### Steps to reproduce the bug:
1. Temporarily delete `app/bundles/CampaignBundle/Tests/DataFixtures/Orm/CampaignData.php`
2. Delete `app/cache/prod` 
3. Backup your database if need be
3. Run `php bin/console mautic:install:data` and enter Y for yes
4. Note the exception that the class could not be found

#### Steps to test this PR:
1. Load up [this PR](https://mautibox.com)
2. Repeat
3. No exceptions are thrown
4. Load up Mautic UI and it should work
